packages/server/src/automations/automationUtils.ts

```diff
@@ -122,17 +122,88 @@ export function getError(err: any) {
 }
 
 export function guardAttachment(attachmentObject: any) {
-  if (
-    attachmentObject &&
-    (!("url" in attachmentObject) || !("filename" in attachmentObject))
-  ) {
+  if (!attachmentObject) {
+    return
+  }
+  if (typeof attachmentObject !== "object") {
+    throw new Error(
+      `Attachments must be objects with both "url" and "filename" keys.`
+    )
+  }
+  if (!("url" in attachmentObject) || !("filename" in attachmentObject)) {
     const providedKeys = Object.keys(attachmentObject).join(", ")
     throw new Error(
       `Attachments must have both "url" and "filename" keys. You have provided: ${providedKeys}`
     )
   }
 }
 
+function deriveFilenameFromUrl(url?: string) {
+  if (!url) return ""
+  try {
+    const pathname = url.startsWith("http") ? new URL(url).pathname : url
+    const parts = pathname.split("/")
+    return parts[parts.length - 1] || ""
+  } catch {
+    return ""
+  }
+}
+
+function normalizeSingleAttachment(input: any): AutomationAttachment | null {
+  if (input == null || input === "") {
+    return null
+  }
+
+  if (typeof input === "string") {
+    try {
+      const parsed = JSON.parse(input)
+      return normalizeSingleAttachment(parsed)
+    } catch {
+      return { url: input, filename: deriveFilenameFromUrl(input) }
+    }
+  }
+
+  if (typeof input !== "object") {
+    throw new Error(`Unsupported attachment value type: ${typeof input}`)
+  }
+
+  const url: string | undefined = input.url
+  if (!url) {
+    const providedKeys = Object.keys(input).join(", ")
+    throw new Error(
+      `Attachments must have both "url" and "filename" keys. You have provided: ${providedKeys}`
+    )
+  }
+  const filename: string =
+    input.filename ?? input.name ?? deriveFilenameFromUrl(url)
```

Comment:
NAB: Just double checking that if `filename` is an empty string here, that you want to use the empty string. 

i.e. you want `??` and not `||`

Reply:
yea, moreso guarding against nulls here.